cpp/src/parquet/page_index_benchmark.cc

```diff
@@ -0,0 +1,153 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <vector>
+
+#include "benchmark/benchmark.h"
+
+#include "parquet/metadata.h"
+#include "parquet/page_index.h"
+#include "parquet/schema.h"
+#include "parquet/test_util.h"
+#include "parquet/thrift_internal.h"
+
+namespace parquet {
+
+void BM_ReadOffsetIndex(::benchmark::State& state) {
+  auto builder = OffsetIndexBuilder::Make();
+  const size_t num_pages = state.range(0);
+  constexpr size_t page_size = 1024;
+  constexpr size_t first_row_index = 10000;
+  for (size_t i = 0; i < num_pages; ++i) {
+    builder->AddPage(page_size * i, page_size, first_row_index * i);
+  }
+  const int64_t final_position = 4096;
+  builder->Finish(final_position);
+  auto sink = CreateOutputStream();
+  builder->WriteTo(sink.get());
+  auto buffer = sink->Finish().ValueOrDie();
+  ReaderProperties properties;
+  for (auto _ : state) {
+    auto offset_index = OffsetIndex::Make(buffer->data() + 0, buffer->size(), properties);
+    ::benchmark::DoNotOptimize(offset_index);
+  }
+  state.counters["num_pages"] = num_pages;
+}
+
+BENCHMARK(BM_ReadOffsetIndex)->Range(8, 1024);
+
+// The sample string length for FLBA and ByteArray benchmarks
+constexpr static uint32_t kDataStringLength = 8;
+
+void GenerateRandomString(uint32_t length, uint32_t seed, std::vector<uint8_t>* heap) {
+  // Character set used to generate random string
+  const std::string charset =
+      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+  std::default_random_engine gen(seed);
+  std::uniform_int_distribution<uint32_t> dist(0, static_cast<int>(charset.size() - 1));
+
+  for (uint32_t i = 0; i < length; i++) {
+    heap->push_back(charset[dist(gen)]);
+  }
+}
+
+template <typename T>
+void GenerateBenchmarkData(uint32_t size, uint32_t seed, T* data,
```

Comment:
this part is ported from `bloom_filter_benchmark.cc` for fast poc. Should I create a `benchmark_util.h` for this?

Reply:
Yes, that sounds reasonable.